src/main/java/com/svydovets/bibirnate/mapper/RegularFieldMapper.java

```diff
@@ -0,0 +1,71 @@
+package com.svydovets.bibirnate.mapper;
+
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Blob;
+import java.sql.Date;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import com.svydovets.bibirnate.exceptions.EntityMappingException;
+import com.svydovets.bibirnate.exceptions.TypeMismatchException;
+
+
+public class RegularFieldMapper implements EntityFieldMapper {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T> void mapField(Field field, T instance, Object value) {
+        try {
+            field.setAccessible(true);
+            value = defineValue(value, field);
+            field.set(instance, value);
+        } catch (IllegalAccessException ex) {
+            throw new EntityMappingException(String.format("Could not access field: %s while mapping entity: %s",
+              field.getName(), instance.getClass().getSimpleName()), ex);
+        } catch (IllegalArgumentException ex) {
+            throw new TypeMismatchException(
+              String.format("Database type: %s doesn't match provided entity type: %s",
+                value.getClass().getSimpleName(), field.getType().getSimpleName()), ex);
+        }
+    }
+
+    private Object defineValue(Object value, Field field) {
+        if (value instanceof Integer intVal
+            && (field.getType().equals(Short.class) || field.getType().equals(short.class))) {
+            value = intVal.shortValue();
+        } else if (value instanceof BigDecimal bdVal && field.getType().equals(BigInteger.class)) {
```

Comment:
why we convert BigDecimal to BigInteger here? maybe we need to throw TypeMismatch exception here, because these types are different?